### PR TITLE
[dualtor][active-active] Add testcase to cover gRPC server failure

### LINKED ISSUE #7027: [test gap][202205][active-active] Validate NiC failure scenario
Add testcase to validate the NiC failure:
Verify that if the NiC changes the mux state arbitrarily, SONiC could correct the NiC's mux state back.

### PR DESCRIPTION
**Pull request by lolyu**
### Description of PR

Summary:
Fixes #7027

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
To verify that SONiC could recover from the arbitrary toggles of the gRPC server.

Signed-off-by: Longxiang Lyu <lolv@microsoft.com>

#### How did you do it?
Add testcase `test_grpc_server_failure.py`

#### How did you verify/test it?
Run on `dualtor-mixed` testbed.
```
dualtor/test_grpc_server_failure.py::test_grpc_server_failure[True] PASSED                                                                                                                                                                                             [ 50%]
dualtor/test_grpc_server_failure.py::test_grpc_server_failure[False] PASSED                                                                                                                                                                                            [100%]
```

### CODE REVIEW

File: tests/dualtor/test_grpc_server_failure.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        logging.debug("Mux status from nic_simulator:\n%s", mux_status)
+        for port in mux_status:
+            if ((mux_status[port][ActiveActivePortID.UPPER_TOR] != upper_tor_forwarding_state) or
+                (mux_status[port][ActiveActivePortID.LOWER_TOR] != lower_tor_forwarding_state)):
+                logging.debug("Port %s mux status is not expected", port)
+                return False
+        return True
+
+    if init_port_state == ForwardingState.STANDBY:
+        pause_icmp_responder(test_mux_ports)
```
Comment (zjswhhh) [inline tests/dualtor/test_grpc_server_failure.py:74]: So the init `standby` state is achieved by pausing icmp responder? 

Just being curious, why don't we `config mux mode` CLI as we already have similar fixture in `nic_simulator_control`? Shouldn't that be easier?

**Review by zjswhhh — APPROVED** [review]
lgtm

File: tests/dualtor/test_grpc_server_failure.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    if not active_active_ports:
+        pytest.skip("Skip as no 'active-active' mux ports available")
+    test_ports = random.sample(active_active_ports, min(len(active_active_ports), TEST_COUNT))
+    logging.info("Select ports %s to test", test_ports)
+    return random.sample(active_active_ports, TEST_COUNT)
+
+
+@pytest.fixture(params=[ForwardingState.ACTIVE, ForwardingState.STANDBY])
+def init_port_state(request):
+    return request.param
```
[the remark this answers is not in the document]
Comment (lolyu) [reply tests/dualtor/test_grpc_server_failure.py:51]: Yes, u r correct.

File: tests/dualtor/test_grpc_server_failure.py (hunk 1 above)
Comment (lolyu) [reply tests/dualtor/test_grpc_server_failure.py:74]: This testcase is to verify SONiC could recover from nic failure in `auto` mode. The one you mentioned should be a different scenario?

**Review by zjswhhh — APPROVED** [review]

**Pull request merged**